torch/linalg/__init__.py

```diff
@@ -1640,23 +1704,26 @@
 
 
 qr = _add_docstr(_linalg.linalg_qr, r"""
-qr(input, mode='reduced', *, out=None) -> (Tensor, Tensor)
+qr(A, mode='reduced', *, out=None) -> (Tensor, Tensor)
 
-Computes the QR decomposition of a matrix. Supports batched inputs.
+Computes the QR decomposition of a matrix.
 
-For a matrix :math:`A`, this is defined as
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+the QR decomposition of a matrix
+:math:`A \in \mathbb{K}^{m \times n}` is defined as
 
 .. math::
 
-    A = QR
+    A = QR\mathrlap{\qquad Q \in \mathbb{K}^{m \times m}, \mathbb{K}^{m \times n}}
 
 where :math:`Q` is orthogonal in the real case and unitary in the complex case, and :math:`R` is upper triangular.
 
-The returned decomposition is a named tuple `(Q, R)`.
-If :attr:`input` is a batch of matrices, then `Q`, `R` are also batched with the same
-batch dimensions.
+Supports inputs of float, double, cfloat and cdouble dtypes.
+Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
+
+The returned decomposition is a named tuple `(Q, R)`
 
-The parameter :attr:`mode` controls the shape of the output. If :attr:`input` has shape
+The parameter :attr:`mode` controls the shape of the output. If :attr:`A` has shape
```

Comment:
Saying the mode parameter controls the shape is a little odd; yes different modes can cause different shapes, but the mathematical intuition behind mode is not that the computation produces different shapes, right?